packages/teams-js/src/internal/communication.ts

```diff
@@ -320,6 +324,45 @@ export function sendMessageToParentAsync<T>(actionName: string, args: any[] | un
   });
 }
 
+/**
+ * @hidden
+ * Send a message to parent requesting a MessageChannel Port.
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export function requestPortFromParentWithVersion(
+  apiVersionTag: string,
+  actionName: string,
+  args: any[] | undefined = undefined,
+): Promise<MessagePort> {
+  if (!isFollowingApiVersionTagFormat(apiVersionTag)) {
+    throw Error(
+      `apiVersionTag: ${apiVersionTag} passed in doesn't follow the pattern starting with 'v' followed by digits, then underscore with words, please check.`,
+    );
+  }
+  return new Promise((resolve) => {
+    const request = sendMessageToParentHelper(apiVersionTag, actionName, args);
+    resolve(waitForPort(request.id));
+  });
```

Comment:
That is a good question - the pattern above is the one followed by all the other functions in the communication.ts file, which is what I based this on:
<img width="681" alt="image" src="https://github.com/OfficeDev/microsoft-teams-library-js/assets/56459261/09e42c83-e1e6-4bfe-84f8-e5fdbbd72e65">

According to Copilot they are probably equivalent, but if we want to change the pattern, I would recommend changing all the instances in the file, rather than just one?

Reply:
Ok, as much as I like consistency, I've agreed to clear improvements on typing and other things that break with the patterns in the file. :) I'll update my function.